File: SubscriptionApi/SubscriptionApi/Authorization/AuthorizationStructure.cs
using System.Collections.Generic;
using SubscriptionApi.Commands;
using SubscriptionApi.ResponseBuilders;

namespace SubscriptionApi.Authorization
{
  public class AuthorizationStructure : 
    AssetAccessAuthorization,
    UserAuthorization, 
    AssetQueryResolution
  {
    public void VerifyAccessTo(string assetName, string userName, AssetAuthorizationEvents events)
    {
      events.NotAuthorizedForAsset(assetName, userName);
    }

    public void VerifyUserExistence(string userName, UserAuthorizationEvents events)
    {
      events.UserNotAuthorized(userName);
    }

    public List<string> RetrieveAssetsByHardwareName(string name)
    {
      throw new System.NotImplementedException();
    }

    public List<string> RetrieveAssetsByUserName(string name)
    {
      throw new System.NotImplementedException();
    }

    public List<string> RetrieveAssetsByOrganizationalUnitName(string name)
    {
      throw new System.NotImplementedException();
    }
  }
}

File: Java/factories/_01_no_factories/src/main/java/Authorization/AuthorizationStructure.java
package Authorization;

import Commands.AssetAccessAuthorization;
import Commands.UserAuthorization;
import ResponseBuilders.AssetAuthorizationEvents;
import ResponseBuilders.UserAuthorizationEvents;

import java.util.List;

public class AuthorizationStructure implements
    AssetAccessAuthorization,
    UserAuthorization,
    AssetQueryResolution {
    public void VerifyAccessTo(String assetName, String userName, AssetAuthorizationEvents events) {
        events.NotAuthorizedForAsset(assetName, userName);
    }

    public void VerifyUserExistence(String userName, UserAuthorizationEvents events) {
        events.UserNotAuthorized(userName);
    }

    public List<String> RetrieveAssetsByHardwareName(String name) {
        throw new RuntimeException("not implemented");

    }

    public List<String> RetrieveAssetsByUserName(String name) {
        throw new RuntimeException("not implemented");
    }

    public List<String> RetrieveAssetsByOrganizationalUnitName(String name) {
        throw new RuntimeException("not implemented");
    }
}
